gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -73,6 +73,7 @@
   private final String endpoint;
   private final String mtlsEndpoint;
   private final String quotaProjectId;
+  private final String apiKey;
```

Comment:
If this is similar to `quotaProjectId` that we are expecting customers to set it, it's better to expose it in `ClientSettings` like [how quotaProjectId did it](https://github.com/googleapis/gax-java/blob/fa85d9cebfee7af57f9792f91f33a6b80711d63a/gax/src/main/java/com/google/api/gax/rpc/ClientSettings.java#L100). Creating a client with ClientSettings is the preferred way compare to Stubs.

Reply:
Thank you Blake for the suggestion. I just added to ClientSettings and update the sample listed in the PR description. Please take another look. Thank you!